compiler/rustc_resolve/src/lib.rs

```diff
@@ -999,7 +999,8 @@ pub struct Resolver<'a> {
 
     /// Table for mapping struct IDs into struct constructor IDs,
     /// it's not used during normal resolution, only for better error reporting.
-    struct_constructors: DefIdMap<(Res, ty::Visibility)>,
+    /// Also includes of list of each fields visibility
+    struct_constructors: DefIdMap<(Res, ty::Visibility, Vec<ty::Visibility>)>,
```

Comment:
```suggestion
    struct_constructors: DefIdMap<(Res, Vec<ty::Visibility>)>,
```
The more detailed field visibilities make the constructor visibility redundant.

Reply:
I am not sure how I'd handle this case: https://github.com/rust-lang/rust/blob/e3ac611b828d579944709e9bf6fe2aede263a7af/compiler/rustc_resolve/src/build_reduced_graph.rs#L970

(also this seems to suggest resolution happens differently across crates? is there a way I can do a ui test with 2 crates?

Reply:
Ah, I see.
Collecting field visibilities here would require decoding them from other crate's metadata, which is better avoided.
It's ok to keep the constructor visibility here then.